tensorflow_io/frame/python/ops/frame_ops.py

```diff
@@ -0,0 +1,168 @@
+    self._data = data
+    self._columns = columns
+    self._index = np.arange(data[columns[0]].shape[0].value)
+    self._iloc = _ILocIndexer(self)
+
+  @property
+  def columns(self):
+    return self._columns
+
+  @property
+  def index(self):
+    return self._index
+
+  @property
+  def values(self):
+    """values"""
+    if len(self._columns) == 1:
+      return self._data[self._columns[0]].numpy()
+    l = [tf.int32, tf.int64, tf.float32, tf.float64, tf.string]
+
+    dtype = l[
+        max([l.index(self._data[column].dtype) for column in self.columns])]
+    if dtype == tf.string:
+      raise NotImplementedError("DataFrame does not support to np.object")
+    raise NotImplementedError("DataFrame does not support values yet")
+
+  @property
+  def shape(self):
+    return len(self.index), len(self.columns)
+
+  @property
+  def iloc(self):
+    return self._iloc
+
+  def __len__(self):
+    return len(self._index)
+
+  def __iter__(self):
+    return iter(self._index)
+
+  def __getitem__(self, item):
+    if isinstance(item, str):
+      if item in self._columns:
+        return DataFrame({item: tf.Variable(self._data[item])}, columns=[item])
+    if isinstance(item, DataFrame):
+      if len(item.columns) != 1:
+        raise NotImplementedError("DataFrame only supports 0-D or 1-D index")
+      if item._data[item.columns[0]].dtype != tf.bool: # pylint: disable=protected-access
+        raise NotImplementedError("DataFrame only supports bool index now")
+      data = {}
+      for column in self._columns:
+        data[column] = tf.Variable(
+            tf.boolean_mask(self._data[column], item._data[item.columns[0]])) # pylint: disable=protected-access
+      return DataFrame(data, list(self._columns))
+    return self._data[item]
+
+  def __setitem__(self, key, value):
+    if key in self.columns:
+      self._data[key] = tf.Variable(value)
+      return
+    raise NotImplementedError("DataFrame only supports setting column")
+
+  def __getattr__(self, name):
+    if name in self._columns:
+      return self.get(name)
+    return object.__getattribute__(self, name)
+
+  def __eq__(self, other):
+    data = dict(
+        (column, tf.Variable(tf.math.equal(self._data[column], other))
+        ) for column in self.columns)
+    return DataFrame(data, columns=list(self.columns))
+
+  def keys(self):
+    return self._columns
+
+  def get(self, key, default=None):
+    try:
+      return DataFrame({key: tf.Variable(self._data[key])}, columns=[key])
+    except (KeyError, ValueError, IndexError):
+      return default
+
+  def pop(self, item):
+    e = self._data.pop(item)
+    self._columns.remove(item)
+    return DataFrame({item: e}, columns=[item])
+
+  def head(self, n=5):
+    return self.iloc[:n]
+
+  def tail(self, n=5):
+    if n == 0:
+      return self.iloc[0:0]
+    return self.iloc[-n:]
+
+  def split(self, func):
+    indices_x, indices_y = func(self._index)
```

Comment:
What if users want to use one of the [splitter classes](https://scikit-learn.org/stable/modules/classes.html#splitter-classes)? Can we support that?

Reply:
@terrytangyuan I think it should be possible, likely by checking the input to see if it is a splitter class or a function. The split could call accordingly.